ipa-core/src/protocol/context/dzkp_validator.rs

```diff
@@ -941,22 +944,21 @@ mod tests {
 
             // gen 1024 random values
             for _i in 0..1024 {
+                // we need to set shares left and right identical since it is a test against a parties own shares
```

Comment:
I'm confused by this. Surely we can just construct 3 batches that correspond to the batches held by each of the 3 parties, and release this constraint that x_left == x_right and so forth.

Reply:
In the actual protocol, we have 3 different parties. In the actual protocol, HP1 right is the same as HP2 left and so on. Here I basically simulate all 3 parties with a single party that has left and right shares that are identical.

There will be more tests that test the protocol setting, I have a different PR that implements the dzkp_multiplication such that values are collected during multiply (https://github.com/private-attribution/ipa/pull/1034). Once both PRs are shipped, I can add a test that runs a multiplication and then checks the prover and verifiers values for consistency.

Reply:
> Here I basically simulate all 3 parties with a single party that has left and right shares that are identical

I think it's really confusing to the reader and sends confusing messages. I also doesn't test any realistic scenario.

I suggest you do something like this:

```
let mut batch_prover = Batch::new(1);
let mut batch_verifier_left = Batch::new(1);
let mut batch_verifier_right= Batch::new(1);
```

Then generate 3 segments from these values (along with the missing 3rd share of each), then at the end call 

```
batch_prover
            .get_field_values_prover::<Fp61BitPrime>()
            .zip(batch_verifier_left.get_field_values_verifier_left::<Fp61BitPrime>())
            .zip(batch_verifier_right.get_field_values_verifier_right::<Fp61BitPrime>())
            .for_each(|((prover, verifier_left), verifier_right)| {
                assert_eq!(prover.0, verifier_left);
                assert_eq!(prover.1, verifier_right);
            });
```